openhcl/virt_mshv_vtl/src/processor/mshv/tlb_lock.rs

```diff
@@ -50,12 +50,17 @@ impl UhProcessor<'_, HypervisorBacked> {
     /// Check the status of the TLB lock of the target VTL on the current VP.
     pub(crate) fn is_tlb_locked(&mut self, requesting_vtl: Vtl, target_vtl: GuestVtl) -> bool {
```

Comment:
Now that this is only called from `debug_assert`, should we protect it behind `cfg(debug_assertions)`? And is there any purpose in adding the caching behavior?